### PR TITLE
[doc][yba] Patching nodes and upgrade notes

### PR DESCRIPTION
**Pull request by ddhodge**
Patching nodes
Notes for upgrading ybdb on universes
DOC-238
DOC-213
DOC-244

@netlify /preview/yugabyte-platform/manage-deployments/upgrade-nodes/

### CODE REVIEW

File: docs/content/preview/yugabyte-platform/manage-deployments/upgrade-software.md [hunk 1]
```diff
@@ -13,32 +13,74 @@ type: docs
...
+
+To view the releases that are available, do the following:
+
+- Click the user profile icon and choose **Releases**.
+
+    ![Releases](/images/yp/releases-list.png)
+
+If a release that you want to install on a universe is not available, import it as follows:
+
+1. On the **Releases** page, click **Import** to open the **Import Releases** dialog shown in the following illustration:
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/manage-deployments/upgrade-software.md:62]:
```suggestion
1. On the **Releases** page, click **Import** to open the **Import Releases** dialog as shown in the following illustration:
```

**Review by aishwarya24 — APPROVED** [review]
LGTM

File: docs/content/preview/manage/upgrade-deployment.md [hunk 2]
```diff
@@ -18,9 +18,27 @@ If you plan to script this in a loop, then a pause of approximately 60 seconds i
 
 Your `data/log/conf` directories are generally stored in a separate location which stays the same across the upgrade so that the cluster data, its configuration settings are retained across the upgrade.
 
+## Before you begin
+
+Before starting the upgrade, review the following major changes to YugabyteDB. Depending on the upgrade you are planning, you may need to make changes to your automation.
+
+### Upgrading to v2.16.0
```
Comment (aishwarya24) [inline docs/content/preview/manage/upgrade-deployment.md:25]: The titles "Upgrading to v2.16.0" and "Upgrading to v2.18.0" seems confusing considering that we are on a preview page.

File: docs/content/preview/yugabyte-platform/manage-deployments/edit-universe.md [hunk 3]
```diff
@@ -9,17 +9,33 @@ menu:
...
-Expansion of universes created with an on-premises cloud provider and secured with third-party certificates obtained from external certification authorities follows a different workflow. For details, see [Expand the universe](../../security/enable-encryption-in-transit#expand-the-universe).
+Using the **Edit Universe** page, you can modify the following:
+
+- Regions - you can select any region configured in the provider
+- Number of nodes and Availability zones - as you add nodes, they are automatically distributed among the availability zones; you can also add, configure, and remove availability zones
+- [Master placement](../../create-deployments/dedicated-master/)
+- Instance type and volume size - you can select instance types configured in the provider
+- [User tags](../instance-tags/)
+
+YugabyteDB Anywhere performs these modifications through the YB-Masters powering the universe. The YB-Masters ensure that the new nodes start hosting the tablet leaders for a set of tablets in such a way that the tablet leader count remains evenly balanced across all the available nodes.
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/manage-deployments/edit-universe.md:30]: link to YB-Masters.

File: docs/content/preview/yugabyte-platform/manage-deployments/upgrade-nodes.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+Typically, the following sequence will be automated using scripts that call the YBA REST APIs.
+
+For each node in the universe, use the following general procedure:
+
+1. Make sure that all nodes in the universe are running correctly.
+
+1. Stop the processes for the node to be patched.
+
+    In YugabyteDB Anywhere, navigate to the universe **Nodes** tab and, for the node, click **Actions** and choose **Stop Processes**.
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/manage-deployments/upgrade-nodes.md:31]:
```suggestion
    In YugabyteDB Anywhere (YBA), navigate to the universe **Nodes** tab and, for the node, click **Actions** and choose **Stop Processes**.
```

File: docs/content/preview/yugabyte-platform/manage-deployments/upgrade-nodes.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    Ensure that the node retains its IP addresses after the patching of the Linux OS. Also ensure that the existing data volumes on the node remain untouched by the OS patching mechanism.
+
+    Two common ways to patch or upgrade the OS of VMs include the following:
+
+    - Inline patching - where you modify the Linux OS binaries in place (for example, yum).
+    - Boot disk replacement - this is typically used with a hypervisor or public cloud. You create a separate new VM with a virtual disk containing the new Linux OS patch or upgrade, disconnect the virtual disk from the new VM, and use it to replace the DB node's boot disk.
+
+1. Start the processes for the node.
+
+    In YugabyteDB Anywhere, navigate to the universe **Nodes** tab and, for the node, click **Actions** and choose **Start Processes**.
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/manage-deployments/upgrade-nodes.md:53]:
```suggestion
    In YBA, navigate to the universe **Nodes** tab and, for the node, click **Actions** and choose **Start Processes**.
```

File: docs/content/preview/yugabyte-platform/manage-deployments/upgrade-software.md [hunk 6]
```diff
@@ -13,32 +13,81 @@ type: docs
 
 The YugabyteDB release that is powering a universe can be upgraded to get new features and fixes included in the release.
 
-Before starting the upgrade:
+## Before you begin
 
-- If you're upgrading a universe from v2.15.x or earlier to v2.16.0 or later, ensure that port 18018 is open on all YugabyteDB nodes so that YB Controller (introduced in v2.16.0) can operate.
+You cannot upgrade a universe to a version of YugabyteDB that is later than the version of YugabyteDB Anywhere. To upgrade to a more recent version of YugabyteDB, you may first have to [upgrade YugabyteDB Anywhere](../../upgrade/).
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/manage-deployments/upgrade-software.md:18]:
```suggestion
You cannot upgrade a universe to a version of YugabyteDB that is later than the version of YugabyteDB Anywhere (YBA). So, to upgrade to a more recent version of YugabyteDB, you may first have to [upgrade YugabyteDB Anywhere](../../upgrade/).
```

File: docs/content/preview/yugabyte-platform/manage-deployments/upgrade-software.md [hunk 7]
```diff
@@ -13,32 +13,81 @@ type: docs
...
-- Consider importing a specific YugabyteDB release into YugabyteDB Anywhere, as follows:
+In addition, before starting the upgrade, review the following major changes in previous YugabyteDB releases. Depending on the upgrade you are planning, you may need to make changes to your automation.
 
-  - Click the user profile icon and select **Releases**.
+Changes in these releases are cumulative. For example, to upgrade from v2.14 to v2.20, your automation will need to take into account the changes in v2.16 and v2.18.
 
-  - Click **Import** to open the **Import Releases** dialog shown in the following illustration:
+### Upgrading from versions earlier than v2.16.0
+
+The YB Controller (YBC) service was introduced in v2.16.0 for all universes (except Kubernetes), and is required for YugabyteDB Anywhere 2.16.0 and later.
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/manage-deployments/upgrade-software.md:26]:
```suggestion
The YB Controller (YBC) service was introduced in v2.16.0 for all universes (except Kubernetes), and is required for YBA 2.16.0 and later.
```

File: docs/content/preview/yugabyte-platform/manage-deployments/upgrade-software.md [hunk 8]
```diff
@@ -13,32 +13,81 @@ type: docs
...
+
+- Firewall ports - update your firewall rules to allow incoming TCP traffic on port 18018, which is used by YBC, for all nodes in a universe.
+
+### Upgrading from versions earlier than v2.18.0
+
+YBC was introduced for Kubernetes clusters in v2.18.0. Refer to [Upgrading to 2.16.0](/#upgrading-to-v2-16-0).
+
+### Upgrading from versions earlier than v2.18.2
+
+The Node Agent was introduced for all universes in v2.18.2. Node agent is an RPC service running on a YugabyteDB node, and is used to manage communication between YugabyteDB Anywhere and the nodes in universes. Except for Day 0 tasks during initial installation, YBA no longer uses SSH and SCP to manage nodes; instead, YBA connects to the Node agent process listening on port 9000, and performs all its management via this secure connection. For more information, refer to the [Node agent FAQ](../../../faq/yugabyte-platform/#node-agent).
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/manage-deployments/upgrade-software.md:47]:
```suggestion
The Node Agent was introduced for all universes in v2.18.2. Node agent is an RPC service running on a YugabyteDB node, and is used to manage communication between YBA and the nodes in universes. Except for Day 0 tasks during initial installation, YBA no longer uses SSH and SCP to manage nodes; instead, YBA connects to the Node agent process listening on port 9000, and performs all its management via this secure connection. For more information, refer to the [Node agent FAQ](../../../faq/yugabyte-platform/#node-agent).
```

File: docs/content/preview/yugabyte-platform/upgrade/_index.md [hunk 9]
```diff
@@ -13,6 +12,8 @@ menu:
 type: indexpage
 ---
 
+Keep YugabyteDB Anywhere up to date for the latest fixes and improvements, and to be able to [upgrade your universes](../manage-deployments/upgrade-software/) to the latest version of YugabyteDB. You cannot upgrade a universe to a version of YugabyteDB that is later than the version of YugabyteDB Anywhere.
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/upgrade/_index.md:15]:
```suggestion
Keep YugabyteDB Anywhere (YBA) up to date for the latest fixes and improvements, and to be able to [upgrade your universes](../manage-deployments/upgrade-software/) to the latest version of YugabyteDB. You cannot upgrade a universe to a version of YugabyteDB that is later than the version of YBA.
```

**Review by aishwarya24 — COMMENTED** [review]
some comments. Also, can you update the upgrade section for [2.18](https://deploy-preview-21099--infallible-bardeen-164bc9.netlify.app/v2.18/yugabyte-platform/upgrade/) similar to stable.
YBA installer was introduced in 2.18, but is missing its upgrade page in 2.18. 
- add an Upgrade YBA installer page in 2.18 upgrade section- https://deploy-preview-21099--infallible-bardeen-164bc9.netlify.app/v2.18/yugabyte-platform/upgrade/
- Update the 2.18 Upgrade section as stable and preview with its structure - https://deploy-preview-21099--infallible-bardeen-164bc9.netlify.app/stable/yugabyte-platform/upgrade/.

File: docs/content/v2.18/yugabyte-platform/upgrade/upgrade-yp-installer.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+To upgrade using YBA Installer, first download the version of YBA Installer corresponding to the version of YBA you want to upgrade to.
+
+Download and extract the YBA Installer by entering the following commands:
+
+```sh
+$ wget https://downloads.yugabyte.com/releases/{{<yb-version version="v2.18" format="long">}}/yba_installer_full-{{<yb-version version="v2.18" format="build">}}-linux-x86_64.tar.gz
+$ tar -xf yba_installer_full-{{<yb-version version="v2.18" format="build">}}-linux-x86_64.tar.gz
+$ cd yba_installer_full-{{<yb-version version="v2.18" format="build">}}/
+```
+
```
[the remark this answers is not in the document]
Comment (aishwarya24) [reply docs/content/v2.18/yugabyte-platform/upgrade/upgrade-yp-installer.md:48]: As a follow-up to https://github.com/yugabyte/yugabyte-db/pull/21153/files

**Review by aishwarya24 — APPROVED** [review]
LGTM

**Review by nkhogen — APPROVED** [review]

**Pull request merged**